drivers/hmis/app/models/hmis/strict_decimal.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Hmis::StrictDecimal < ActiveModel::Type::Decimal
+  def cast(value)
+    return if value.nil?
+
+    raise ArgumentError, "Invalid decimal value: #{value.inspect}" if value.is_a?(String) && value !~ /\A[+-]?\d+(\.\d+)?\z/
+
+    super
```

Comment:
Surprisingly, the seems happy enough to cast non-String values as well:
```
irb(main):026> value = {"foo" => "bar" }; ib.TotalMonthlyIncome = value; ib.save!; ib.TotalMonthlyIncome.to_s
=> "0.0"
```

I don't think the code here will address that since the value is not a string?

Maybe we should adjust it to 
```
raise if value.to_s !~ DECIMAL_REGEX
```